samples/bluetooth/df_per_adv_beacon/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+cmake_minimum_required(VERSION 3.13.1)
+
+if (NOT DF_MODE)
+  message("DF_MODE is not being defined on the CMake command-line in the environment or by the app\n"
+          "Setting DF_MODE to default value: AOD")
+  set(DF_MODE "AOD")
+endif()
+
+if (DF_MODE STREQUAL "AOA")
+  set(DF_CONF_FILE "${CMAKE_CURRENT_SOURCE_DIR}/df_aoa.conf")
+elseif (DF_MODE STREQUAL "AOD")
+  set(DF_CONF_FILE "${CMAKE_CURRENT_SOURCE_DIR}/df_aod.conf")
+else()
+  message(FATAL_ERROR "Selected unsupported DF_MODE: ${DF_MODE}.\n"
+          "Please set DF_MODE parameter to one of following: AOA, AOD.")
+endif()
+
+if(NOT EXISTS ${DF_CONF_FILE})
+  message(FATAL_ERROR "No such file or directory: DF_CONF_FILE: '${DF_CONF_FILE}'")
+endif()
+list(APPEND CONF_FILE ${DF_CONF_FILE})
+list(APPEND CONF_FILE "prj.conf")
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+
+project(df_beacon)
+
+target_sources(app PRIVATE
+  src/main.c
+)
+
+if (CONFIG_BT_CTLR_DF_ANT_SWITCH_TX)
+set(DTC_OVERLAY_FILE "${CMAKE_CURRENT_SOURCE_DIR}/boards/nrf52833dk_nrf52833.overlay")
```

Comment:
I've added it by use of condition to show that information provided by nrf52833dk_nrf52833.overlay are required only for Direction Finding in AoD mode (when antennas are switched). For simple beacons working in AoA mode the overlay is not needed.

Reply:
@tejlmand have you got an opinion about recommended practice in this regard?

Reply:
Unless I am missing something, this is not correct. Adding to DTC_OVERLAY_FILE at this point in the build will have no effect. You need to set DTC_OVERLAY_FILE before `find_package(Zephyr)`, since DTC_OVERLAY_FILE is processed by boilerplate.cmake during the `find_package()` and is ignored afterwards.

I understand this is a chicken-and-egg problem since you want to handle this after Kconfig, which runs after dtc.

I see no way to resolve this within our current build system constraints except for documentation.

Reply:
> @tejlmand have you got an opinion about recommended practice in this regard?

No, this is not a correct way.
What @mbolivar-nordic  describes is fully correct, but also remember that placing an overlay in `<sample>/boards` folder means it will automatically be sourced when building for that board, regardless of the Kconfig setting.
 
So when exactly is `nrf52833dk_nrf52833.overlay` needed ?
1. Always when building for the `nrf52833dk_nrf52833` board ?
2. Always when `CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=y` regardless of the board in use ?
3. Only when building for the `nrf52833dk_nrf52833` board and `CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=y` ?

Reply:
> You need to set DTC_OVERLAY_FILE before `find_package(Zephyr)`, since DTC_OVERLAY_FILE is processed by boilerplate.cmake during the `find_package()` and is ignored afterwards.

FYI, I've sent a pull request which will detect and warn on invalid settings like this: https://github.com/zephyrproject-rtos/zephyr/pull/32556

Reply:
> but also remember that placing an overlay in `<sample>/boards` folder means it will automatically be sourced when building for that board, regardless of the Kconfig setting.

Good point -- I missed that. So it has no effect, but it's redundant :)

Reply:
@mbolivar-nordic @tejlmand thank you very much for your input!

> So when exactly is nrf52833dk_nrf52833.overlay needed ?
At the moment it is needed for only when building for  `nrf52833dk_nrf52833` board and CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=y
But it will change in future, when DF support is added to other boards. Then the overlay might be shared between all those boards.

Reply:
> At the moment it is needed for only when building for nrf52833dk_nrf52833 board and CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=y
But it will change in future, when DF support is added to other boards. Then the overlay might be shared between all those boards.

in that case, we could also do it this way:
```
west build -b nrf52833dk_nrf52833 samples/bluetooth/df_per_adv_beacon -- -DCONF_FILE=prj_aod.conf
```
and then place the overlay specific for the build type: `boards/ nrf52833dk_nrf52833_aod.overlay`

But that would swap `aoa` <-> `aod` as to which is the default configuration.
And require this pr also: https://github.com/zephyrproject-rtos/zephyr/pull/32436

See more about this here: https://docs.zephyrproject.org/latest/guides/kconfig/setting.html#the-initial-configuration

Reply:
@tejlmand But I can't wait until the PR: https://github.com/zephyrproject-rtos/zephyr/pull/32436 is merged... My PR must go not later than tomorrow. I need it to prepare another PR in ncs/zephyr that must go by the Friday.

Reply:
@tejlmand This is last piece that is missing here. What do you think about leaving the `nrf52833dk_nrf52833_aod.overlay` to be included into the build independently from `CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=y` and update the sample when PR: https://github.com/zephyrproject-rtos/zephyr/pull/32436 is ready?